### ISSUE TITLE
Ngx-Echarts Animation not working Initially

### ISSUE THREAD

**Issue by KrisnaPrashatt** [xieziyu/ngx-echarts#413]
I am using ngx echarts for my angular app and I am not able to get the animation to work during the initial load of the chart. I have attached a sample [Stackblitz](https://stackblitz.com/edit/angular-ngx-echarts-tfgkan?file=src%2Fapp%2Fapp.component.html,src%2Fapp%2Fapp.component.ts,package.json) where you can observe that on initial load, the bars render faster whereas when toggling using the legend, the actual animation happens.

Can anyone help me out in the right direction saying what have I done wrong?

Html:

`<div echarts [options]="options"></div>`

Ts options object

`{
  xAxis: {
    data: xAxisData,
  },
  legend: {},
  yAxis: {},
  series: [
    {
      name: 'bar',
      type: 'bar',
      barCategoryGap: '0%',
      data: data1,
      animationDuration: 5000,
    },
    {
      name: 'bar1',
      type: 'bar',
      barCategoryGap: '0%',
      data: data2,
      animationDuration: 3000,
    },
  ],
}`

**Comment by xieziyu** [comment]
@KrisnaPrashatt 
ECharts applies the transition animation only when data changes after chart initialized. So you can assign an empty value to `option` firstly in order to initialize the chart; then, in `ngAfterViewInit`, you can assign data to your chart option using `setTimeout`. For example:

```ts
option: EChartsOption = {}

ngAfterViewInit()
{
  setTimeout(() => {
    this.option = {
       // options and series...
    };
  );
}
```

**Comment by ArielCW** [comment]
@xieziyu Could the current fix on version 17 be applied to **16.2.0** ? or some kind of workaround to keep the auto resize and the initial animation on version 16?

**Comment by s9ke** [comment]
As a workaround, you could probably just add the property `animationDurationUpdate` to the options with a defined time like the following [stackblitz](https://stackblitz.com/edit/angular-ngx-echarts-jrxzgs?file=src%2Fapp%2Fapp.component.ts).

Here is the documentation on [ECharts](https://echarts.apache.org/en/option.html#animationDurationUpdate).
